squid/entrypoint.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
```

Comment:
Where does this file come from? Ideally we'd get it from some upstream git repo where we can also get updates

Reply:
Got this by execing into the only trustworthy image as the upstream: https://hub.docker.com/r/ubuntu/squid/tags

Seems not to have any Dockerfile https://github.com/chainguard-dev/image-requests/issues/1556#issuecomment-2020435125

Reply:
Re: this part:

```bash
# re-create snakeoil self-signed certificate removed in the build process
if [ ! -f /etc/ssl/private/ssl-cert-snakeoil.key ]; then
    /usr/sbin/make-ssl-cert generate-default-snakeoil --force-overwrite > /dev/null 2>&1
fi
```

snakeoil is what debian uses as the name for it's default self-signed cert I believe. So not sure this transitions well here to wolfi. We might not even need this if we already have one or take care of it in existing packaging? This part probably needs a bit more digging into

Reply:
Yes, even I am not sure about it, I feel our `openssl` package could be sufficient but I will dig more in this

Reply:
I have removed this snakeoil thing and would like to test this by building an image so would love to know if this can be approved and we can see what works or is needed more?

Reply:
Having discussed it with Jamon, we both feel that the current state of entrypoint.sh looks good with removal of cert generation from the script. Additionally, in order to run this as non-root, this PR adds opts to build with default user as squid

Reply:
I'm ok with this file as it is - there aren't likely to be any changes since it is just tailing logs and running squid with argv passed through.